tools/lib/NQP/Configure/Macros.pm

```diff
@@ -90,15 +94,64 @@ sub expand {
     return $text_out;
 }
 
-sub _m_test {
-    #my $self = shift;
-    return "*this is a test* ($_[1])";
+sub split_spaced {
+    my $self = shift;
+
+    return split /(?<!\\)\s/s, shift;
+}
+
+# include(file1 file2)
+# Include a file. Parameter is expanded first, then the result is used a the
+# file name. File content is been expanded.
+# Multiple filenames are split by spaces. If file path contains a space in it it
+# must be quoted with \
+sub _m_include {
+    my $self = shift;
+    my @filenames = $self->split_spaced( $self->expand(shift) );
+    my $text = "";
+
+    for my $file ( map { $self->_m_unescape($_) } @filenames ) {
+        next unless $file; # Split may result in some empty items.
+        die "Circular dependency detected on including $file" 
+            if $self->{'-including'}{$file};
+        $self->{'-including'}{$file} = 1;
+        $text .= "\n## Included from $file\n";
+        $text .= $self->expand( NQP::Configure::slurp( $file ) );
+        $text .= "\n## End of included $file\n"; 
+        delete $self->{'-including'}{$file};
+    }
+    return $text;
 }
 
+# sp_escape(a string)
+# Escapes all spaces in a string with \
+# Implicitly called by @@ macros
 sub _m_sp_escape {
-    my $str = $_[1];
+    my $self = shift;
+    my $str = self->expand($_[1]);
     $str =~ s{ }{\\ }g;
     $str;
 }
 
+# unescape(a\ st\ring)
+# Simlpe unescaping from backslashes. Replaces any \<char> sequence with <char>
+sub _m_unescape {
+    my $self = shift;
+    my $str = shift;
+    $str =~ s/\\(.)/$1/g;
+    return $str;
+}
+
+# nfs(dir/file)
```

Comment:
It's `nfp` and it's actually already changed. I thought github autoupdate PR when branch is updated.

Reply:
It did. I went through your commits cronologically because I found that easier to understand than looking at the resulting diff. I made the comment when I first saw that change and didn't yet know you already fixed it in a later commit.